cloud/ovirt/ovirt_storage_domains.py

```
@@ -0,0 +1,433 @@
+    )
+    check_sdk(module)
+
+    try:
+        connection = create_connection(module.params.pop('auth'))
+        storage_domains_service = connection.system_service().storage_domains_service()
+        storage_domains_module = StorageDomainModule(
+            connection=connection,
+            module=module,
+            service=storage_domains_service,
+        )
+
+        state = module.params['state']
+        control_state(storage_domains_module)
+        if state == 'absent':
+            ret = storage_domains_module.remove(
+                destroy=module.params['destroy'],
+                format=module.params['format'],
+                host=module.params['host'],
+            )
+        elif state == 'present':
+            sd_id = storage_domains_module.create()['id']
+            storage_domains_module.post_create_check(sd_id)
+            ret = storage_domains_module.action(
+                action='activate',
+                action_condition=lambda s: s.status == sdstate.MAINTENANCE,
+                wait_condition=lambda s: s.status == sdstate.ACTIVE,
+                fail_condition=failed_state,
+            )
+        elif state == 'maintenance':
+            sd_id = storage_domains_module.create()['id']
+            storage_domains_module.post_create_check(sd_id)
+            ret = storage_domains_module.action(
+                action='deactivate',
+                action_condition=lambda s: s.status == sdstate.ACTIVE,
+                wait_condition=lambda s: s.status == sdstate.MAINTENANCE,
+                fail_condition=failed_state,
+            )
+        elif state == 'unattached':
+            ret = storage_domains_module.create()
+            storage_domains_module.pre_remove(
+                storage_domain=storage_domains_service.service(ret['id']).get()
+            )
+            ret['changed'] = storage_domains_module.changed
+
+        module.exit_json(**ret)
+    except Exception as e:
+        module.fail_json(msg=str(e))
```

Same here, please surface the full exception.